geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentFeatureSerializer.scala

```diff
@@ -70,38 +70,45 @@ class ConfluentFeatureSerializer(
   }
 
   override def deserialize(id: String, bytes: Array[Byte]): SimpleFeature = {
-    val record = kafkaAvroDeserializer.get.deserialize("", bytes).asInstanceOf[GenericRecord]
+    val record = kafkaAvroDeserializers.get.deserialize("", bytes).asInstanceOf[GenericRecord]
 
-    val feature = featureReader.get.read(id, record)
+    val feature = featureReaders.get.read(id, record)
 
     // set the feature visibility if it exists
-    Option(sft.getUserData.get(GeoMesaAvroVisibilityField.KEY)).map(_.asInstanceOf[String]).foreach { fieldName =>
-      try {
-        SecurityUtils.setFeatureVisibility(feature, record.get(fieldName).toString)
-      } catch {
-        case NonFatal(ex) => throw new IllegalArgumentException(s"Error setting feature visibility using" +
```

Comment:
Is there no longer any chance that an exception is thrown?

Reply:
I think there could be, but i think the main causes here would be if the vis field was null in the record, and i added a check for that. any exceptions would get bubbled up, instead of being wrapped as they were, but the end result should be similar.